src/mpid/ch4/src/ch4_types.h

```diff
@@ -67,6 +67,7 @@ typedef struct MPIDIG_hdr_t {
     int tag;
     MPIR_Context_id_t context_id;
     int error_bits;
+    uint8_t flags;
```

Comment:
Maybe we can use bit-fields here? It'll be nice to keep this header 16-byte.

Reply:
I don't think this would be a problem. I will move the `tag` `context_id` and `error_bits` to a new `MPIDIG_send_msg_t`. Those are send specific fields and the common `MPIDIG_hdr_t` header would only have the fields commonly needed all messages that transmit data.